extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleMetadataOverrides.java

```diff
@@ -45,7 +46,7 @@ public final class OracleMetadataOverrides {
      *
      * @param reflectiveClass buildItem producer
      */
-    @BuildStep
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
     void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
```

Comment:
I would expect these to be executed only when in native as `ReflectiveClassBuildItem` is only consumed in native.